src/js/select2/data/select.js

```diff
@@ -14,15 +14,23 @@ define([
 
   SelectAdapter.prototype.current = function (callback) {
     var data = [];
-    var self = this;
 
-    this.$element.find(':selected').each(function () {
-      var $option = $(this);
+    if (this.$element.length > 0) {
+      var domElement = this.$element[0];
 
-      var option = self.item($option);
+      var selectedElements = domElement.querySelectorAll(':checked');
 
-      data.push(option);
-    });
+      for (var index in selectedElements) {
+        if (selectedElements.hasOwnProperty(index)) {
```

Comment:
I suspect this is allowing non-DOMElement properties of the `NodeList` to make their way through in tests which is causing the test failures we are seeing in CI. I would recommend falling back to a traditional index-based `for` loop which should be fine for this situation and should ensure that we only ever try to call `.item` on a DOM element.